procountor/client.py

```diff
@@ -115,11 +115,29 @@ def _get_token(self):
             'content-type': 'application/x-www-form-urlencoded'
         }
 
-        r = requests.post(self.api_url + 'oauth/token/', params=params, headers=headers)
+        url = self.api_url + 'oauth/token/'
+        response = requests.post(url, params=params, headers=headers)
+        status_code = response.status_code
 
-        self.access_token = r.json()['access_token']
+        if status_code != 200:
 
-        return self.access_token
+            if status_code == 404:
+                raise RuntimeError("Not found api endpoint. Please, check your API version.")
+
+            if status_code == 401:
+                raise RuntimeError("Authentication failed. Please, check your credentials.")
+
+            raise RuntimeError("Authentication got an unexpected HTTP Status code: " + status_code + ".")
+
+        json_content = response.json()
+
+        access_token = json_content.get('access_token', None)
+
+        if access_token is None:
+            raise RuntimeError("Cannot read the access_token from the response.")
```

Comment:
maybe add the json_content to the exception, makes it easier to see what was returned?